pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/PythonExtension.groovy

```diff
@@ -67,15 +67,13 @@ class PythonExtension {
     public Map<String, Map<String, String>> forcedVersions = [
         'argparse'      : ['group': 'pypi', 'name': 'argparse',         'version': '1.4.0'],
         'flake8'        : ['group': 'pypi', 'name': 'flake8',           'version': '2.5.4'],
-        'pbr'           : ['group': 'pypi', 'name': 'pbr',              'version': '1.8.0'],
```

Comment:
The fact that we install in graph post-order.

Now when a package that really has pbr in `setup_requires` gets installed, pbr will sniff the virtualenv, but all the required dependencies will already be there and pbr will be happy.

With previous alphabetical order, most of the dependencies would be installed late (e.g., psutils) although they are needed early in the dependency graph. In such cases, pbr would immediately bail out complaining about missing dependencies. The workaround was to install pbr itself early when virtualenv has only base setup requirements and no missing dependencies. The downside was that we could not upgrade setuptools beyond 19.1.1 and pbr was artificially made a dependency of **every** package.

Now, we can remove it, and it will be brought in on-demand only for packages that really need it. Also, we can finally upgrade our setup requirements (setuptools, virtualenv, ...) to support Python 3.6 properly.